lib/src/main/kotlin/com/swmansion/starknet/provider/Provider.kt

```diff
@@ -321,6 +321,17 @@ interface Provider {
 
     fun getTransactionStatus(transactionHash: Felt): Request<GetTransactionStatusResponse>
 
+    /**
+     * Get L1 handler transaction data.
+     *
+     * Get L1 handler transaction data for all L1 → L2 messages sent by the given L1 transaction.
+     *
+     * @param l1TransactionHash The hash of the L1 transaction.
+     *
+     * @throws RequestFailedException
+     */
+    fun getMessagesStatus(l1TransactionHash: NumAsHex): Request<GetMessagesStatusResponse>
```

Comment:
This should return an array:
https://github.com/starkware-libs/starknet-specs/blob/647caa00c0223e1daab1b2f3acc4e613ba2138aa/api/starknet_api_openrpc.json#L269